prepare_build

```diff
@@ -33,6 +33,19 @@ mkdir -p $BUILD_DIR/gen
 # Enter build directory
 cd $BUILD_DIR
 
+# MacOS-specific installations
+if [[ "$OSTYPE" == "darwin"* ]]; then
+  if [[ ! $(command -v brew) ]]; then
+    BREW_INSTALL="https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh"
+    /bin/bash -c "$(curl -fsSL $BREW_INSTALL)"
+  fi
+
+  brew install cmake
+  brew install coreutils
+  brew install vcpkg
+  brew install pkg-config
+fi
+
```

Comment:
In the "Pre-requisites" section above, we say that cmake should be installed by the user first.  Here we are installing it automatically.  If we install automatically here for mac, does it make sense to do the same on other platforms?  Although it gets trickier to support Debian and RH distributions.

Also, this script installs vcpkg in the build folder (see lines 50-53 below).  We shouldn't need to install via brew.

At a higher level, what do users expect when working with this repo?  Do they expect it to make global changes to their machine, like installing brew, coreutils, vcpkg, etc?

By not installing those things everything that this repo modifies on disk is contained within the cloned directory.  When you're done with the SDK, you can simply delete the directory and everything is gone from your machine.  Is that a cleaner model, although it might require users to manually perform some pre-requisite steps?

Reply:
Good questions :)
I think our options are
1- Making a second script to install those dependencies (cmake, pkg-config, etc.)
2- Including those commands in the README
3- Updating what the script is meant to do and add the install steps to prepare_build

I think I interpreted b/232103609 as meaning 3, but I guess an extra README section (2) might be more appropriate. Ultimately I don't know what the exact preference is for what the script does so I'm OK with any options.